node/src/utils/counting_channel.rs

```diff
@@ -0,0 +1,194 @@
+//! Support for counting channels.
+//!
+//! Regular tokio channels do not make the number of pending items available. Counting channels wrap
+//! regular tokio channels but keep a counter of the number of items up-to-date with every `send`
+//! and `recv`. The `len` method can be used to retrieve the number of items.
+//!
+//! The channel also counts the heap memory used by items on the stack if `DataSize` is implemented
+//! for `T` and `send_datasized` is used instead of send. Internally it stores the size of each item
+//! instead of recalculating on `recv` to avoid underflows due to interior mutability.
+
+#![allow(dead_code)]
```

Comment:
Can this be removed?

Reply:
Not at this point, as `len` and `send` are unusued. I do expect them to be used for additional metrics in the future and would rather not remove them for this reason.

I can put the `dead_code` allowance onto the methods instead?

Also, keeping `send` around is debateable, that one can probably do.

Reply:
I'd prefer the scope reduced to just the two methods then please, preferably annotated with TODOs :)

Reply:
Alright, can do. I'll remove `send` as well.